boilerplate/app/navigators/navigation-utilities.tsx

```diff
@@ -1,37 +1,35 @@
-import { useState, useEffect, useRef } from "react"
+import React, { useState, useEffect, useRef } from "react"
 import { BackHandler } from "react-native"
 import {
-  PartialState,
-  NavigationState,
-  NavigationAction,
   createNavigationContainerRef,
+  NavigationContainerRef,
+  Route,
 } from "@react-navigation/native"
 
-/* eslint-disable */
-export const RootNavigation = {
-  navigate(_name: string, _params?: any) {},
-  goBack() {},
-  resetRoot(_state?: PartialState<NavigationState> | NavigationState) {},
-  getRootState(): NavigationState {
-    return {} as any
-  },
-  dispatch(_action: NavigationAction) {},
-}
-/* eslint-enable */
-
 export const navigationRef = createNavigationContainerRef()
 
 /**
- * Gets the current screen from any navigation state.
+ *
+ * Gets the current route from any navigation state.
+ * returns an array with [route.key || route.name, route]
+ * route.name is a string, route is the entire route object.
+ * #### Example
+ * ```js
+ * const [currentRouteName, route] = getRoute(navigationRef)
+ * setCurrentScreen(currentRouteName)
+ * ```
  */
-export function getActiveRouteName(state: NavigationState | PartialState<NavigationState>) {
-  const route = state.routes[state.index]
-
-  // Found the active route -- return the name
-  if (!route.state) return route.name
-
-  // Recursive call to deal with nested routers
-  return getActiveRouteName(route.state)
+export const getRoute = (
+  navigationRef: React.MutableRefObject<NavigationContainerRef<any> | null>,
+): [routeName: string | null, route: Route<string, any> | null] => {
+  if (navigationRef.current == null) {
```

Comment:
Are you using `==` instead of `===` here on purpose, to capture `undefined` and `null` both?

We may want an `isEmpty` function that does this, e.g.:

```suggestion
  if (isEmpty(navigationRef.current)) {
```

I'm not sure if `navigationRef` would be `null` or `undefined` by default here.